### DISCUSSION TITLE
Can type extensions be setup in code?

### DISCUSSION THREAD

**Discussion by BaileyMillerSSI** [ChilliCream/hotchocolate#4372] [Q&A]
Background:
I was playing around with stitching today and think that it could be a powerful feature to help build out a micro-graph service.
In one of the examples, I setup I have a user graph and an address graph and then a gateway to tie that together.
From what I understand the only way to add extended properties is use `AddTypeExtensionsFromFile` with the extended properties file.

Question:
Is it possible now or in the future to add type extension on a class? Currently I had to do something like this:
```
extend type User {
    address: Address @delegate(schema: "addresses", path: "addressForUser(userId: $fields:userId)")
}
```
on my gateway server.

However, it would be much nicer if this type of extension could be performed in code.
```
public class UserTypeExtension : ExtendedType
{
  protected override void ExtendType(Object builder)
  {
     builder.SetName("User");
     builder.AddDelegate(schema: "addresses", type: "Address", path: "addressForUser(userId: $fields:userId)");
  }
}

...

services.AddGraphQLServer().AddTypeExtension<UserTypeExtension>()
```

The gateway wouldn't need to know the User class or the Address class but could in code setup the extended properties.

**Comment by michaelstaib** [comment]
Depends on your approach ... if you do it in the gateway then yes. With V13 we are bringing this capability to the downstream services so that you have a full code-first experience for stitching.
